#!/usr/bin/env python

# set up parameters that we care about
PACKAGE = 'jsk_pcl_ros'

try:
    import imp
    imp.find_module(PACKAGE)
    from dynamic_reconfigure.parameter_generator_catkin import *;
except:
    import roslib; roslib.load_manifest(PACKAGE)
    from dynamic_reconfigure.parameter_generator import *;

from math import pi

gen = ParameterGenerator ()

gen.add("plane_angle_threshold", double_t, 0, "angle threshold", 10.0 / 180.0 * pi, 0.0, pi)
gen.add("plane_distance_threshold", double_t, 0, "distance threshold", 0.05, 0.0, 2.0)
gen.add("grid_map_angle_threshold", double_t, 0, "grid map angle threshold", 20.0 / 180.0 * pi, 0.0, pi)
gen.add("grid_map_distance_threshold", double_t, 0, "grid map distance threshold", 0.05, 0.0, 2.0)
gen.add("distance_threshold", double_t, 0, "distance threshold to check collision", 0.01, 0.0, 0.1)
gen.add("collision_check_sampling_d", double_t, 0, "sampling distance of the polygon when checking collision", 0.01, 0.0, 0.5)
gen.add("resolution_size", double_t, 0, "resolution of occupancy grid", 0.05, 0.0, 0.5)
gen.add("required_vote", int_t, 0, "required voting num", 5, 0, 100)
gen.add("static_generation", int_t, 0, "generation to be regard as static map", 10, 0, 100)
exit (gen.generate (PACKAGE, "jsk_pcl_ros", "EnvironmentPlaneModeling"))
